lib/superstore/types/boolean_type.rb

```diff
@@ -2,19 +2,14 @@ module Superstore
   module Types
     class BooleanType < BaseType
       TRUE_VALS = [true, 'true', '1']
-      FALSE_VALS = [false, 'false', '0', '', nil]
-      VALID_VALS = TRUE_VALS + FALSE_VALS
+      FALSE_VALS = [false, 'false', '0']
 
-      def encode(bool)
-        unless VALID_VALS.include?(bool)
-          raise ArgumentError.new("#{bool.inspect} is not a Boolean")
+      def typecast(value)
+        if TRUE_VALS.include?(value)
+          true
+        elsif FALSE_VALS.include?(value)
+          false
         end
-
-        TRUE_VALS.include?(bool)
-      end
-
-      def decode(str)
-        TRUE_VALS.include?(str)
       end
```

Comment:
booleans are properly represented in JSON and therefore encode/decode does not need to be implemented.